### DISCUSSION TITLE
Calculate BLS Signature

### DISCUSSION THREAD

**Discussion by LautaroLasorsa** [Consensys/gnark#699] [Q&A]
Hello everyone,

I am trying to implement a random test for a BLS circuit. 

For that, I need to get a random Hash, a random generator G1 and a random Sk, and with that calculate the Signature and the Public Key.

For the first part, I use random strings, but i am not sure if it is correct.

For the second part, I dont know what I have to do.

Thanks in advance for the help.

**Comment by ivokub** [comment]
Have a look at https://github.com/ConsenSys/gnark/blob/develop/std/algebra/emulated/sw_bn254/doc_test.go#L91.

**Reply by LautaroLasorsa** [reply]
Thanks a lot, that looks very useful
